gulpfile.js

```diff
@@ -0,0 +1,58 @@
+const del = require('del');
+const gulp = require('gulp');
+const ts = require('gulp-typescript');
+const mocha = require('gulp-mocha');
+
+const paths = {
+    BUILD_DIR:  'dist',
+    CLIENT_DIR: 'src/client/public',
+    TESTS_DIR:  'test'
+};
+
+const serverTs = ts.createProject('tsconfig.json');
+const clientTs = ts.createProject(`${paths.CLIENT_DIR}/tsconfig.json`);
+
+gulp.task('transpile-server', () => {
+    return serverTs.src()
+        .pipe(serverTs())
+        .js.pipe(gulp.dest(paths.BUILD_DIR));
+});
+
+gulp.task('transpile-client', () => {
+    return clientTs.src()
+        .pipe(clientTs())
+        .js.pipe(gulp.dest(`${paths.BUILD_DIR}/src/client/public/js`));
+});
+
+gulp.task('copy-html', () => {
+    return gulp.src([`${paths.CLIENT_DIR}/*.html`])
+        .pipe(gulp.dest(`${paths.BUILD_DIR}/${paths.CLIENT_DIR}`));
+});
+
+gulp.task('copy-fixtures', () => {
+    return gulp.src([`${paths.TESTS_DIR}/fixtures/*.json`])
```

Comment:
Also, I'm not fond of copying test data after each build. Need to think about the project structure.

Reply:
Solution: use ts-node.